### PR TITLE
Siemens Vision600 Hoffman

### PR DESCRIPTION
**Pull request by KrisThielemans**

### CODE REVIEW

File: SIRF_data_preparation/run_BSREM.py [hunk 1]
```diff
@@ -28,6 +31,8 @@
 # logging.basicConfig(level=logging.INFO)
 
 scanID = args['<data_set>']
+num_updates = args['--updates']
```
Comment (casperdcl) [inline SIRF_data_preparation/run_BSREM.py:34]:
```suggestion
num_updates = int(args['--updates'])
```

File: SIRF_data_preparation/get_penalisation_factor.py [hunk 2]
```diff
@@ -40,7 +40,8 @@
         print("Need to set the --dataset arguments")
         exit(1)
     if args["--write_penalisation_factor"] is not None:
```
Comment (casperdcl) [inline SIRF_data_preparation/get_penalisation_factor.py:42]:
```suggestion
```

File: SIRF_data_preparation/get_penalisation_factor.py [hunk 3]
```diff
@@ -40,7 +40,8 @@
...
-        write_penalisation_factor = False
+        write_penalisation_factor = args["--write_penalisation_factor"]
```
Comment (casperdcl) [inline SIRF_data_preparation/get_penalisation_factor.py:43]:
```suggestion
    write_penalisation_factor = args["--write_penalisation_factor"]
```

File: SIRF_data_preparation/create_Hoffman_VOIs.py [hunk 4]
```diff
@@ -47,24 +47,28 @@
     if scanID is None:
         print("Need to set the --dataset argument")
         exit(1)
-    if args["--skip_write_PETRIC_VOIs"] is not None:
+    if (args["--skip_write_PETRIC_VOIs"] is not None) and args["--skip_write_PETRIC_VOIs"]:
```
Comment (casperdcl) [inline SIRF_data_preparation/create_Hoffman_VOIs.py:50]:
```suggestion
    if args["--skip_write_PETRIC_VOIs"]:
```

**Review by casperdcl — CHANGES_REQUESTED** [review]

File: SIRF_data_preparation/create_Hoffman_VOIs.py (hunk 4 above)
Comment (KrisThielemans) [reply SIRF_data_preparation/create_Hoffman_VOIs.py:50]: I guess I did some experiments with docopt to come up with this. I don't quite understand why/when docopt sets the value to False if not present, but I trust you of course.

**Comment by KrisThielemans** [comment]
I first ran 10000 updates, but that didn't feel safe enough, so continued with smaller stepsize
```
python -m SIRF_data_preparation.run_BSREM Siemens_Vision600_Hoffman --updates=10000
python -m SIRF_data_preparation.run_BSREM Siemens_Vision600_Hoffman --updates=6000 --initial_image=output/Siemens_Vision600_Hoffman/BSREM/iter_final.hv --initial_step_size=.01 --relaxation_eta=.03 
```
Pass index is 7680 (was close to that when using image at 10000 updates as reference)

# objectives
Note jump after 10000 due to restart
Siemens_Vision600_Hoffman_BSREM_objectives
Siemens_Vision600_Hoffman_BSREM_objectives_last
# metrics
Siemens_Vision600_Hoffman_metrics_BSREM
Siemens_Vision600_Hoffman_metrics_BSREM_cont
Siemens_Vision600_Hoffman_metrics_BSREM_OSEM
# "pass" images
Siemens_Vision600_Hoffman_image_at_0 01_0 005
diff with ref
Siemens_Vision600_Hoffman_ref_diff_image_at_0 01_0 005

**Pull request merged**